File: C/exercise/isBalanced.c

#include "isBalanced.h"
#include <string.h>

bool isBalanced(char *num)
{
	int sum_odd = 0;
	int sum_even = 0;
	int len = strlen(num);
	for (int i = 0; i < len; i += 2) {
		sum_odd += num[i] - '0';
	}
	for (int i = 1; i < len; i += 2) {
		sum_even += num[i] - '0';
	}
	return sum_odd == sum_even;
}

File: Ruby/solution/is_balanced.rb
# @param {String} num
# @return {Boolean}
def is_balanced(num)
  odd = 0
  sum = 0
  num.chars.each_with_index do |c, i|
    v = c.ord - "0".ord
    odd += v if i % 2 == 0
    sum += v
  end
  sum == (odd * 2)
end
